src/modules/workspaces/workspaces.usecase.ts

```diff
@@ -513,6 +521,145 @@ export class WorkspacesUsecases {
     return teamsWithMemberCount;
   }
 
+  async getTeamMembers(teamId: WorkspaceTeam['id']) {
+    const members = await this.teamRepository.getTeamMembers(teamId);
+    return members;
+  }
+
+  async editTeamData(teamId: WorkspaceTeam['id'], editTeamDto: EditTeamDto) {
+    const [team, workspace] = await Promise.all([
+      this.teamRepository.getTeamById(teamId),
+      this.workspaceRepository.findOne({ defaultTeamId: teamId }),
+    ]);
+
+    if (!team || workspace?.isDefaultTeam(team)) {
```

Comment:
The `workspace?.isDefaultTeam(team)` check seems to be irrelevant. If the query finds it, it's the default team, if not, could be due to 2 reasons: 
- The workspace is not the team's, thus, we need to prevent the update
- The workspace is the team's one

If instead of looking at the workspace by the default team you search it by the `team.workspaceId`, things are easier to check:

```
const team = await this.teamRepository.getTeamById(teamId);
const workspace = await this.workspaceRepository.findById(team.workspaceId);

if (!workspace) {
  // someone is doing something nasty -> Forbidden
}

if (!team || workspace.isDefaultTeam(team)) {
  // bad request
}

/*
If you are here, the team is part of the workspace but it is not the default one, safe to continue. Non-existent workspaces or teams and non-related workspaces cases are handled properly
*/
```